modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java

```diff
@@ -2266,10 +2267,18 @@ private String _getSegmentsAnonymousUserId() {
 	}
 
 	private long[] _getSegmentsEntryIds() {
-		return _segmentsEntryRetriever.getSegmentsEntryIds(
-			_themeDisplay.getScopeGroupId(), _themeDisplay.getUserId(),
-			_requestContextMapper.map(
-				PortalUtil.getHttpServletRequest(_portletRequest)));
+		long[] requestSegmentsEntryIds = (long[])_portletRequest.getAttribute(
+			SegmentsWebKeys.SEGMENTS_ENTRY_IDS);
+		long[] portletSegmentsEntryIds =
+			_segmentsEntryRetriever.getSegmentsEntryIds(
+				_themeDisplay.getScopeGroupId(), _themeDisplay.getUserId(),
+				_requestContextMapper.map(
+					PortalUtil.getHttpServletRequest(_portletRequest)));
+
+		return ArrayUtil.toLongArray(
+			SetUtil.fromArray(
+				ArrayUtil.append(
+					requestSegmentsEntryIds, portletSegmentsEntryIds)));
```

Comment:
Could return ArrayUtil.append(requestSegmentsEntryIds, portletSegmentsEntryIds) instead?

Reply:
I think we need to use this for avoiding duplicated values